runtime/src/main/java/org/corfudb/runtime/view/replication/AbstractReplicationProtocol.java

```diff
@@ -61,4 +66,19 @@ public ILogData read(RuntimeLayout runtimeLayout, long globalAddress) {
      * @param globalAddress The address to hole fill.
      */
     protected abstract void holeFill(RuntimeLayout runtimeLayout, long globalAddress);
+
+    /**
+     * Update per-stream compaction mark in the runtime.
+     * @param runtime         corfu runtime
+     * @param compactionMarks compaction marks
+     */
+    protected void updateCompactionMarks(CorfuRuntime runtime, Map<UUID, Long> compactionMarks) {
+        StreamsView sv = runtime.getStreamsView();
+        compactionMarks.forEach((streamId, compactionMark) -> {
+            boolean updateSuccess = sv.get(streamId).updateCompactionMark(compactionMark);
```

Comment:
StreamView will only cache the thread-safe streams, however VLO uses unsafe streams, so using sv.get() is incorrect here.

Reply:
outdated.